addons/project/models/project_task_recurrence.py

```diff
@@ -79,11 +79,28 @@ def _get_recurrence_delta(self):
             f"{self.repeat_unit}s": self.repeat_interval
         })
 
+    def _copy_task_dependency(self, task_copy, task_original):
+        if 'task_mapping' not in self.env.context:
+            self = self.with_context(task_mapping=dict())
```

Comment:
i'm really not a big fan of using the context like that to build the dict. It would be better to pass an empty dict as an argument, and slowly build it up with the return value of each function.